R/plotResource.R

```diff
@@ -0,0 +1,152 @@
+#'   frame with the four variables 'w', 'value', 'Predator', 'Resource' is returned.
+#' @export
+#' @family plotting functions
+#' @seealso [plotting_functions]
+#' @examples
+#' \donttest{
+#' plotResourcePred(NS_params)
+#'
+#' # Returning the data frame
+#' fr <- plotResourcePred(NS_params, return_data = TRUE)
+#' str(fr)
+#' }
+plotResourcePred <- function(object, proportion = TRUE, return_data = FALSE)
+{
+    if (is(object, "MizerSim")) {
+        params <- object@params
+        params <- setInitialValues(params, object)
+    } else if (is(object, "MizerParams")) {
+        params <- validParams(object)
+    }
+    SpIdx <- factor(params@species_params$species,
+                    levels = params@species_params$species)
+
+    #How many backgrounds are being used? Assuming using only mizerMR for now for additional backgrounds
+    if(!is.null(getComponent(params, "MR")))
+    {
+        plot_dat <- NULL
+        for(iRes in 1:dim(params@other_params$other$MR$resource_params)[1])
+        {
+            select <- (params@other_params$MR$capacity[iRes,] > 0)
+            pred_rate <- params@other_params$MR$interaction[,iRes] *
+                getPredRate(params)[, select]
+            total <- colSums(pred_rate)
+
+            ylab <- "Death rate [1/year]"
+            if (proportion) {
+                pred_rate <- pred_rate / rep(total, each = dim(pred_rate)[[1]])
+                ylab = "Proportion of predation"
+            }
+            # Make data.frame for plot
+            plot_dat <- rbind(plot_dat,
+                              data.frame(
+                                  w = rep(params@w_full[select], each = dim(pred_rate)[[1]]),
+                                  value = c(pred_rate),
+                                  Predator = SpIdx,
+                                  Resource = params@other_params$other$MR$resource_params$resource[iRes]
+                              ))
+        }
+        pl <- plotDataFrame(plot_dat, params, style = "area", xtrans = "log10",
+                            xlab = "Resource size [g]", ylab = ylab,
+                            wrap_var = "Resource", wrap_scale = "free")
+    } else {
+        select <- (params@cc_pp > 0)
+        pred_rate <- params@species_params$interaction_resource *
+            getPredRate(params)[, select]
+        total <- colSums(pred_rate)
+        ylab <- "Death rate [1/year]"
+        if (proportion) {
+            pred_rate <- pred_rate / rep(total, each = dim(pred_rate)[[1]])
+            ylab = "Proportion of predation"
+        }
+        # Make data.frame for plot
+        plot_dat <- data.frame(
+            w = rep(params@w_full[select], each = dim(pred_rate)[[1]]),
+            value = c(pred_rate),
+            Predator = SpIdx
+        )
+        pl <- plotDataFrame(plot_dat, params, style = "area", xtrans = "log10",
+                            xlab = "Resource size [g]", ylab = ylab)
+    }
+
+    if(return_data) return(plot_dat) else return(pl)
```

Comment:
As commented previously, move this to earlier so that if the user only wants the data, they don't have to wait for the plots to be created.